homeassistant/components/dlna_dms/config_flow.py

```diff
@@ -0,0 +1,177 @@
+        if not discovery_service_list:
+            return self.async_abort(reason="not_dms")
+        discovery_service_ids = {
+            service.get("serviceId")
+            for service in discovery_service_list.get("service") or []
+        }
+        if not DmsDevice.SERVICE_IDS.issubset(discovery_service_ids):
+            return self.async_abort(reason="not_dms")
+
+        # Abort if another config entry has the same location, in case the
+        # device doesn't have a static and unique UDN (breaking the UPnP spec).
+        self._async_abort_entries_match({CONF_URL: self._location})
+
+        self.context["title_placeholders"] = {"name": self._name}
+
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Allow the user to confirm adding the device."""
+        LOGGER.debug("async_step_confirm: %s", user_input)
+
+        if user_input is not None:
+            return self._create_entry()
+
+        self._set_confirm_only()
+        return self.async_show_form(step_id="confirm")
+
+    def _create_entry(self) -> FlowResult:
+        """Create a config entry, assuming all required information is now known."""
+        LOGGER.debug(
+            "_async_create_entry: location: %s, USN: %s", self._location, self._usn
+        )
+        assert self._name
+        assert self._location
+        assert self._usn
+
+        data = {CONF_URL: self._location, CONF_DEVICE_ID: self._usn}
+        return self.async_create_entry(title=self._name, data=data)
+
+    async def _async_parse_discovery(
+        self, discovery_info: ssdp.SsdpServiceInfo
+    ) -> None:
+        """Get required details from an SSDP discovery.
+
+        Aborts if a device matching the SSDP USN has already been configured.
+        """
+        LOGGER.debug(
+            "_async_parse_discovery: location: %s, USN: %s",
+            discovery_info.ssdp_location,
+            discovery_info.ssdp_usn,
+        )
+
+        if not discovery_info.ssdp_location or not discovery_info.ssdp_usn:
+            raise AbortFlow("bad_ssdp")
+
+        if not self._location:
+            self._location = discovery_info.ssdp_location
+
+        self._usn = discovery_info.ssdp_usn
+        await self.async_set_unique_id(self._usn)
+
+        # Abort if already configured, but update the last-known location
+        self._abort_if_unique_id_configured(
+            updates={CONF_URL: self._location}, reload_on_update=False
+        )
+
+        self._name = (
+            discovery_info.upnp.get(ssdp.ATTR_UPNP_FRIENDLY_NAME)
+            or urlparse(self._location).hostname
+            or DEFAULT_NAME
+        )
+
+    async def _async_get_discoveries(self) -> list[ssdp.SsdpServiceInfo]:
+        """Get list of unconfigured DLNA devices discovered by SSDP."""
+        LOGGER.debug("_get_discoveries")
```

Comment:
The method is called `_async_get_discoveries`. Personally I'd remove the logs that don't log any data.